src/main/java/com/amihaiemil/zold/RestfulZoldWts.java

```diff
@@ -76,4 +75,39 @@ public RestfulZoldWts(final HttpClient client, final URI baseUri) {
         this.client = client;
         this.baseUri = baseUri;
     }
+
+    /**
+    * Pull the wallet from the network.
+    */
+    public void pull() {
+
+    }
+
+    /**
+    * Get the balance of the wallet.
+    * @return Balance
+    */
+    public double balance() {
+        return 0.0;
+    }
+
+    /**
+    * Pay to another wallet.
+    * @param keygap Sender keygap
+    * @param user Recipient user id
+    * @param amount Amount to be sent
+    * @param details The details of transfer
+    */
+    public void pay(String keygap, String user, double amount, String details) {
+
+    }
+
+    /**
+    * Finds all payments that match this query and returns.
+    * @param id Wallet id
+    * @param details Regex of payment details
+    */
+    public void find(final String id, final String details) {
+
+    }
```

Comment:
@ammaratef45 ballance(), pay(...) and find(...) should all be methods of ``Wallet`` :)